### PR TITLE
First version of Michel Tag

### PR DESCRIPTION
**Pull request by juliensnz**

### CODE REVIEW

File: src/Release/Command/TagCommand.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $fs = new Filesystem();
+
+        $io->text(file_get_contents(getcwd(). '/src/Release/resources/welcome.txt'));
+
+        $io->ask('Are you ready?', 'Hell yeah!');
```
Comment (mickaelandrieu) [inline src/Release/Command/TagCommand.php]: :D

File: src/Release/Command/TagCommand.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+        $io->section('Check of access rights');
+
+        $client = new Client();
+        $client->authenticate($this->config['github_token'], Client::AUTH_URL_TOKEN);
```
Comment (mickaelandrieu) [inline src/Release/Command/TagCommand.php]: can return an exception if you reach the limit of accesses in minutes, should be catched imo

File: src/Release/Helper/PullRequestHelper.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                'The modifications on the repository %s/%s are ready to be pushed:',
+                $owner,
+                $repository
+            ),
+            $diff,
+            sprintf('You can review them in the folder %s', $editionFolder)
+        ]);
+
+        if (!$this->io->confirm('Do you want to proceed and create the pull request?', true)) {
+            return false;
```
Comment (mickaelandrieu) [inline src/Release/Helper/PullRequestHelper.php]: false or 0?

File: src/Release/Helper/PullRequestHelper.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        if ($this->io->confirm(
+            sprintf(
+                'We just created a pull request on %s:' .
+                "\n" .
+                '%s' .
+                "\n" .
+                'Do you want to open it in your default browser to check that everything is allright?',
+                $pullRequest['head']['repo']['full_name'],
+                $pullRequest['html_url']
```
Comment (mickaelandrieu) [inline src/Release/Helper/PullRequestHelper.php]: good idea !

File: src/Release/Helper/PullRequestHelper.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                'Merge pull request #%s: Unable to merge the pull request. With message %s',
+                $pullRequest['number'],
+                $e->getMessage()
+            ));
+
+            return false;
+        }
+
+        $this->io->success(sprintf(
+            'Merge pull request #%s',
```
Comment (mickaelandrieu) [inline src/Release/Helper/PullRequestHelper.php]: I guess it's more important to get the pull request label than the number: what do you think?

File: app/console [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+$config = Yaml::parse(file_get_contents(__DIR__.'/../config.yml'));
+
+$application = new Application('michel-tag', '1.0.0');
+$command = new TagCommand($config);
+
+$application->add($command);
```
Comment (juliensnz) [inline app/console]: There will be new command to update the doc and demo after this one :)

File: src/Release/Command/TagCommand.php (hunk 2 above)
Comment (juliensnz) [inline src/Release/Command/TagCommand.php]: If this error occurs, it needs to crash and end the command. What would be the benefit to catch it? Custom message?

File: src/Release/Helper/PullRequestHelper.php (hunk 3 above)
Comment (juliensnz) [inline src/Release/Helper/PullRequestHelper.php]: false as it's used by the caller

File: src/Release/Helper/PullRequestHelper.php (hunk 5 above)
Comment (juliensnz) [inline src/Release/Helper/PullRequestHelper.php]: The label is auto generated and will always be the same. I prefer to use the number

File: app/console (hunk 6 above)
Comment (mickaelandrieu) [inline app/console]: Ok 👍

File: src/Release/Command/TagCommand.php (hunk 2 above)
Comment (mickaelandrieu) [inline src/Release/Command/TagCommand.php]: Yes, something like "You reach your GitHub token limit". I've created a command some months ago to get the list of contributors form a repository, and the exception message returned was everything *except* something useful  :(

File: src/Release/Helper/NameProvider.php [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+     *
+     * @return string
+     */
+    protected function getTagFromUser($reason)
+    {
+        do {
+            $tagName = $this->io->ask(sprintf(
+                '%s. ' .
+                "\n" .
+                'How do you whant to name the tag? (example 1.7.3) ',
```
Comment (shadypierre) [inline src/Release/Helper/NameProvider.php]: want*

File: src/Release/Helper/FileUpdater.php [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+            $this->fs->dumpFile(sprintf('%s/%s', $editionFolder, sprintf(self::CHANGELOG_PATTERN, $branch)), $changelogFile);
+            ProcessRunner::runCommand(sprintf('cd %s && git add %s', $editionFolder, sprintf(self::CHANGELOG_PATTERN, $branch)));
+
+            //We copy the pim_parameters file into the standard
+            $pimParamsFile = $this->downloadRepoFile($owner, $devRepository, $branch, self::PIM_PARAMS, $editionFolder);
+            $this->fs->dumpFile(sprintf('%s/%s', $editionFolder, self::PIM_PARAMS), $pimParamsFile);
+            ProcessRunner::runCommand(sprintf('cd %s && git add %s', $editionFolder, self::PIM_PARAMS));
+        }
+
+        if ('dev' === $repositoryInformations['version'] && 'community' === $repositoryInformations['edition']) {
```
Comment (shadypierre) [inline src/Release/Helper/FileUpdater.php]: Version and edition should be a constant IMO

**Pull request merged**